server/src/main/java/org/opensearch/transport/NativeMessageHandler.java

```diff
@@ -92,7 +98,7 @@ public class NativeMessageHandler implements ProtocolMessageHandler {
         TransportKeepAlive keepAlive
     ) {
         this.threadPool = threadPool;
-        this.outboundHandler = outboundHandler;
+        this.outboundHandler = new NativeOutboundHandler(nodeName, version, features, statsTracker, threadPool, bigArrays, outboundHandler);
```

Comment:
This is basically the whole idea - if we use `NativeMessageHandler`, it means that the inbound message uses `native` protocol, so we could use `NativeOutboundHandler` here